clients/go/apiban-ipsets/apiban-ipsets.go

```diff
@@ -203,6 +209,16 @@ func main() {
 
 	log.Print("Interval for checking the list:", apiconfig.TICK)
 
+	// generate encryption key from passphrase
+	if len(apiconfig.PASSPHRASE) > 0 {
+		if ipcipher, err = anonymization.NewPassphraseCipher(apiconfig.PASSPHRASE); err != nil {
+			log.Fatalln("Cannot initialize ipcipher. Exiting.")
+		}
+		// initialize a validator using the configured passphrase; neither length nor salt are used since this validator verifies only the remote code
+		if validator, err = anonymization.NewPassphraseValidator(apiconfig.PASSPHRASE, 0 /*length*/, "" /*salt*/); err != nil {
```

Comment:
probably a silly question but how can we validate the checksum when the validation gets only the password as input without the UID? Do we take the UID from the event's UDI:HMAC pair? Does it mean we calculate it for every event?  If so, wouldn't calculating it once and doing `memcmp` only to validate be sufficient?